### PR TITLE
[SAI] Refactor SAI and add new features to be compatible with sonic-sairedis 

### PR DESCRIPTION
**Pull request by kcudnik**
Many changes:

* Move utils to it's own namespace dash::utils 
* Move fixed api implementation to proper DashSai class
* Remove unused methods
* Add workaround to return not supported status on get api 
* Remove underscore from variable naming
* Fix route bulk api set/get warnings
* Simplify insertInTable method
* Change asserts to return error codes and add missing switch api
* Print oids in hex form

### CODE REVIEW

**Comment by kcudnik** [comment]
it should be better to review one commit at a time

File: dash-pipeline/SAI/templates/dashsai.cpp.j2 [hunk 1]
```diff
@@ -0,0 +1,649 @@
...
+ *
+ * Name of environment variable, when set to any value, will return proper
+ * SAI_STATUS_NOT_SUPPORTED on switch/port GET api.
+ *
+ * Currently because of limitation of saithrift, GET on switch/port is
+ * returning SAI_STATUS_SUCCESS on not implemented attribute.
+ *
+ * TODO FIXME needs to be addressed in saithrift.
+ *
+ * This is temporary workaround for syncd, it should be removed when fixed in saithrift.
```
Comment (chrispsommers) [inline dash-pipeline/SAI/templates/dashsai.cpp.j2:28]: Is there a corresponding issue filed against saithrift?

File: dash-pipeline/SAI/templates/saiapi.cpp.j2 [hunk 2]
```diff
@@ -509,6 +513,35 @@ static sai_status_t dash_sai_get_{{ table.name }}_attribute(
     assert(0 && "sai_get_{{ table.name }}_attribute NYI");
     return SAI_STATUS_FAILURE;
 }
+
+{% if table.name == 'route_entry' %}
```
Comment (chrispsommers) [inline dash-pipeline/SAI/templates/saiapi.cpp.j2:517]: Why this particular case?

**Review by chrispsommers — APPROVED** [review]
LGTM

File: dash-pipeline/SAI/templates/dashsai.cpp.j2 (hunk 1 above)
Comment (kcudnik) [reply dash-pipeline/SAI/templates/dashsai.cpp.j2:28]: not yet

**Pull request merged**